src/traffic_server/HostStatus.cc

```diff
@@ -94,6 +94,30 @@ mgmt_host_status_down_callback(ts::MemSpan span)
   }
 }
 
+static void
+handle_record_read(const RecRecord *rec, void *edata)
+{
+  HostStatus &hs = HostStatus::instance();
+  std::string hostname;
+  std::string reason;
+
+  if (rec) {
+    // parse the hostname from the stat name
+
+    char *s = const_cast<char *>(rec->name);
+    // 1st move the pointer past the stat prefix.
+    s += strlen(stat_prefix.c_str());
+    hostname = s;
+    reason   = hostname.substr(hostname.find('_'));
+    reason.erase(0, 1);
+    // erase the reason tag
+    hostname.erase(hostname.find('_'));
+    if (rec->data.rec_int == 0 && Reasons::validReason(reason.c_str())) {
```

Comment:
Can you specify that you are trying to look for hosts that were down. Also, How are we going to track time for hosts that were marked down for a certain period of time?